src/typeDefs/inputs.ts

```diff
@@ -339,6 +339,8 @@ export const inputs = gql`
     lastName: String
     maritalStatus: MaritalStatus
     phone: UserPhoneInput
+    applangcode: String
+    id: ID
```

Comment:
remove the `id` field from here, the update will only be applied for a user with `_id === context.userId`

Reply:
if this mutation is to be allowed to be triggered by both normal users and superadmins, some changes have to be made:-

1. if `args.input.id` is not provided(it is `null` or `undefined`), the update will be carried out on the user with `user._id` equals to `context.userId`

1. if `args.input.id` is provided, there are two scenarios:-
    1. if it's a normal user, `context.userId` must be equal to `args.input.id`
    2. if it's a superadmin there are no restrictions, they can update any user with `user._id` equals to `args.input.id`

some things to confirm with @palisadoes :-
1. can one superadmin update other superadmin's user data?
2. is this allowed only for superadmins, or admins of organizations too? the thing to remember is that a user is not tied to any one organization, they can be members/admins of many organizations simultaneously, **_this functionality doesn't exist in talawa-api right now because of existence of `userType === "ADMIN"` field on a user object, when in actuality the relationship between a user and an organization should exist on a junction between them, joining table in sql it is called_**

Reply:
1. @xoldyckk is correct on the userType to organization relationship. This should be fixed in this issue.
    1. https://github.com/PalisadoesFoundation/talawa-api/issues/1711
1. Senior usertypes should be able to edit or demote users of junior rank. They should be able to promote junior ranks to no more than their rank. Super Admins will need privileges to do this for everyone. I'm broadly basing this on Google's guidelines for their apps.
    1. https://apps.google.com/supportwidget/articlehome?hl=en&article_url=https%3A%2F%2Fsupport.google.com%2Fa%2Fanswer%2F1219251%3Fhl%3Den&assistant_event=welcome&assistant_id=usermasterbot&product_context=1219251&product_name=UnuFlow&trigger_context=a

Reply:
Hey @xoldyckk,

I included the `id` in the mutation because during the API call to edit a specific user, in the backend `context.userId` only contained the `id` of the logged-in user. As a result, the profile being updated was that of the current user rather than the intended one. To address this, I passed the `id` of the user that requires editing/updating in the API call.

Following this logic, if `args.input.id` is `null`, it indicates the root user; otherwise, it's the `id` of the user whose profile needs updating.

Reply:
ok. then it looks fine.

Reply:
@Devesh326 take some time and contemplate about what i said previously, the current logic as it stands allows a normal user to update user fields of any other user, they just need to pass in the other user's `id` in the input, frontend application can't prevent normal users from making these requests

understand the authentication/authorization flows, self review your code and possible edge cases, only then ask for reviews from other contributers

Reply:
Yes noted.